lock/common_test.go

```diff
@@ -0,0 +1,293 @@
+package lock
+
+import (
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type LockSuite struct {
+	suite.Suite
+	NewService func() Service
+}
+
+func (s *LockSuite) TestLockUnlockSingleLockNoConcurrency() {
+	assert := s.Assert()
+	niter := 100
+	id := "mylock"
+	service := s.NewService()
+	cfg := &SessionConfig{TTL: 1 * time.Second}
+	session, err := service.NewSession(cfg)
+	assert.NoError(err)
```

Comment:
That's right. However, so far I've been using this style of instantiating `t`, `assert` and `require` at the beginning of the test. This has the advantage of the test being easier to convert between non-suite and suite tests.